apis/externalsecrets/v1alpha1/secretstore_vault_types.go

```diff
@@ -72,6 +79,13 @@ type VaultProvider struct {
 	// +kubebuilder:default:="v2"
 	Version VaultKVStoreVersion `json:"version"`
 
+	// Parser defines how backend data is interpreted by engine.
+	// Parser defaults to "vk".
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum="kv";"string"
+	// +kubebuilder:default:="kv"
+	Parser VaultParser `json:"parser,omitempty"`
```

Comment:
As you cannot use key/value structure directly without refactoring code i added Parser parameter to store config to make backwards compatible solution.